### PR TITLE
Add Callhome to lambdas, Heap.io

### PR DESCRIPTION
**Pull request by danielmkn**
Heap.io calls added to all lambdas, except for Authorizer.
Added an actual list of supported regions. 
Minor bug fixes.

### CODE REVIEW

File: eventProcessor/app.js [hunk 1]
```diff
@@ -24,13 +27,45 @@ const lambdaInvoke = (issuesChunks) => {
...
 
+const axiosClient = axios.create({
+  baseURL: 'https://heapanalytics.com/api',
+  headers: {
+    'Content-Type': 'application/json',
+    accept: 'application/json',
+  }
+});
+
+const sendCallHomeData = async (callHomePayload) => {
```
Comment (alexhung) [inline eventProcessor/app.js:38]: This is ok for now. We should extract this into its own file/module similar to logger.

File: eventProcessor/app.js [hunk 2]
```diff
@@ -40,11 +75,27 @@ export async function lambdaHandler(event) {
...
+        xray_issues_received: event.length,
+        messages_sent_to_issue_processor: results.filter((result) => (result.status === 'fulfilled')).length,
+        failed_messages: results.filter((result) => (result.status === 'rejected')).length,
+        action: 'process-xray-payload-security-hub',
+        jpd_url: `https://${hostName}`
+      }
+      await sendCallHomeData(callHomePayload);
+      logger.info(`HeapIO request has been sent.`);
+    } catch (e) {
+      logger.info(`Error while sending info to HeapIO. ${e}`);
```
Comment (alexhung) [inline eventProcessor/app.js:96]: Use `logger.warn`

File: issueProcessor/app.js [hunk 3]
```diff
@@ -53,11 +55,60 @@ const sendSQSmessage = async (event) => {
...
+        region: SECURITY_HUB_REGION,
+        xray_issues_received: event.length,
+        messages_sent_to_sqs: results.Successful.length,
+        action: 'send-issue-to-sqs-security-hub',
+        jpd_url: `https://${event[0].host_name}`
+      }
+      await sendCallHomeData(callHomePayload);
+      logger.info(`HeapIO request has been sent.`);
+    } catch (e) {
+      logger.info(`Error while sending info to HeapIO. ${e}`);
```
Comment (alexhung) [inline issueProcessor/app.js:109]: Use `logger.warn`

File: template.yaml [hunk 4]
```diff
@@ -32,25 +32,26 @@ Parameters:
...
       - eu-west-1
       - eu-west-2
-      - eu-south-1
       - eu-west-3
-      - eu-north-1
       - me-south-1
       - sa-east-1
+      - us-east-1
       - us-east-2
+      - us-gov-east-1
```
Comment (alexhung) [inline template.yaml:53]: No GovCloud, please remove these.

File: transformer/app.js [hunk 5]
```diff
@@ -195,29 +196,65 @@ const sendUpdateCommand = async (updateBody) => {
...
+      identity: callHomePayload.jpd_url,
+      event: 'transform-issue-and-send-to-security-hub',
+      properties: callHomePayload,
+    };
+
+    logger.info('Sending data to Heap.io, path: /track', { body });
+
+    response = await axiosClient.post('/track', body);
+  } catch (e) {
+    logger.error('Failed to send data to Heap.io', { error: e.toJSON() });
```
Comment (alexhung) [inline transformer/app.js:240]: No need for `toJSON()`. The logger will stringify the `e` object. Also `toJSON()` is specific to Axios function.

File: transformer/app.js [hunk 6]
```diff
@@ -277,6 +315,23 @@ export async function lambdaHandler(event, context) {
...
+        xray_issues_imported: successfulFindings.length,
+        xray_issues_updated: hubUpdateResponse.SuccessCount,
+        xray_issues_import_failed: failedFindingsIDs.length,
+        action: 'transform-issue-and-send-to-security-hub',
+        jpd_url: `https://${issue.host_name}`
+      }
+      await sendCallHomeData(callHomePayload);
+      logger.info(`HeapIO request has been sent.`);
+    } catch (e) {
+      logger.info(`Error while sending info to HeapIO. ${e}`);
```
Comment (alexhung) [inline transformer/app.js:332]: Use `logger.warn`

File: transformer/app.js [hunk 7]
```diff
@@ -195,29 +196,65 @@ const sendUpdateCommand = async (updateBody) => {
...
+  } catch (e) {
+    logger.error('Failed to send data to Heap.io', { error: e.toJSON() });
+  }
+  return response;
+}
+
+
 export async function lambdaHandler(event, context) {
   let response;
+  let issue;
```
Comment (alexhung) [inline transformer/app.js:248]: Let's initialize this to empty object, to be on safe side.

**Review by alexhung — CHANGES_REQUESTED** [review]
Some minor updates needed

File: eventProcessor/app.js (hunk 1 above)
Comment (danielmkn) [reply eventProcessor/app.js:38]: will do in the next PR

File: eventProcessor/app.js (hunk 2 above)
Comment (danielmkn) [reply eventProcessor/app.js:96]: fixed

File: issueProcessor/app.js (hunk 3 above)
Comment (danielmkn) [reply issueProcessor/app.js:109]: fixed

File: template.yaml (hunk 4 above)
Comment (danielmkn) [reply template.yaml:53]: Both removed.

File: transformer/app.js (hunk 5 above)
Comment (danielmkn) [reply transformer/app.js:240]: Removed.

File: transformer/app.js (hunk 6 above)
Comment (danielmkn) [reply transformer/app.js:332]: Fixed.

File: transformer/app.js (hunk 7 above)
Comment (danielmkn) [reply transformer/app.js:248]: Ok.

**Review by alexhung — APPROVED** [review]

**Pull request merged**